engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala

```diff
@@ -3,23 +3,25 @@ package pl.touk.nussknacker.engine.compile
 import cats.data.Validated.{invalid, valid}
 import cats.data.ValidatedNel
 import pl.touk.nussknacker.engine.compile.ProcessCompilationError.{MissingParameters, NodeId, RedundantParameters}
-import pl.touk.nussknacker.engine.definition.DefinitionExtractor.ObjectDefinition
 
 object Validations {
 
-  def validateParameters[T>:PartSubGraphCompilationError](definedParamNames: List[String], usedParamNames: List[String])(implicit nodeId: NodeId)
+  def validateParameters[T >: PartSubGraphCompilationError <: ProcessCompilationError](definedParamNames: List[String],
+                                                                                       usedParamNames: List[String],
+                                                                                       parametersUsedForOtherPurpose: Set[String])
+                                                                                      (implicit nodeId: NodeId)
   : ValidatedNel[T, Unit]= {
+    val syntax = ValidatedSyntax[T]
+    import syntax._
+
     val usedParamNamesSet = usedParamNames.toSet
     val definedParamNamesSet = definedParamNames.toSet
-    val redundantParams = usedParamNamesSet.diff(definedParamNamesSet)
+    val redundantParams = usedParamNamesSet.diff(definedParamNamesSet ++ parametersUsedForOtherPurpose)
```

Comment:
where are parametersUsedForOtherPurpose used? Couldn't track where it is sth other than Set.empty ;)